hedera-mirror-rest-java/src/main/java/com/hedera/mirror/restjava/controller/AllowancesController.java

```diff
@@ -0,0 +1,115 @@
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+@CustomLog
+@RequiredArgsConstructor
+@RequestMapping("/api/v1/accounts/{id}/allowances")
+@RestController
+public class AllowancesController {
+
+    private final NftAllowanceService service;
+    private final NftAllowanceMapper accountMapper;
+
+    @CrossOrigin(origins = "*")
+    @GetMapping(value = "/nfts")
+    NftAllowancesResponse getNftAllowancesByAccountId(
+            @PathVariable EntityIdParameter id,
+            @RequestParam(name = ACCOUNT_ID, required = false) EntityIdRangeParameter accountId,
+            @RequestParam(defaultValue = DEFAULT_LIMIT) @Positive @Max(MAX_LIMIT) int limit,
+            @RequestParam(defaultValue = "asc") Sort.Direction order,
+            @RequestParam(defaultValue = "true") boolean owner,
+            @RequestParam(name = TOKEN_ID, required = false) EntityIdRangeParameter tokenId) {
+
+        NftAllowanceRequestBuilder requestBuilder = NftAllowanceRequest.builder()
+                .accountId(id)
+                .isOwner(owner)
+                .limit(limit)
+                .order(order)
+                .ownerOrSpenderId(accountId)
+                .tokenId(tokenId);
+
+        var serviceResponse = service.getNftAllowances(requestBuilder.build());
+
+        NftAllowancesResponse response = new NftAllowancesResponse();
+        response.setAllowances(accountMapper.map(serviceResponse));
+
+        String next = null;
+        if (!serviceResponse.isEmpty() && serviceResponse.size() == limit) {
+            next = buildNextLink(owner, order, response, serviceResponse);
+        }
+        response.links(new Links().next(next));
+        return response;
+    }
+
+    private static String buildNextLink(
+            boolean owner,
+            Sort.Direction order,
+            NftAllowancesResponse response,
+            Collection<NftAllowance> serviceResponse) {
+
+        HashMap<String, String> lastValues = new HashMap<>();
+        HashMap<String, Boolean> included = new HashMap<>();
+        if (response.getAllowances() != null) {
+            if (owner) {
+                lastValues.put(
+                        ACCOUNT_ID,
+                        response.getAllowances().get(serviceResponse.size() - 1).getSpender());
+
+            } else {
+                lastValues.put(
+                        ACCOUNT_ID,
+                        response.getAllowances().get(serviceResponse.size() - 1).getOwner());
+            }
+            included.put(ACCOUNT_ID, true);
+            lastValues.put(
+                    TOKEN_ID,
+                    response.getAllowances().get(serviceResponse.size() - 1).getTokenId());
+            included.put(TOKEN_ID, false);
+        }
+        return Utils.getPaginationLink(false, lastValues, included, order);
+    }
```

Comment:
Can be greatly simplified, I believe. We might be able to get rid of included map if we make it a linked map and assume last item is not inclusive?

```suggestion
        var last = CollectionUtils.lastElement(response.getAllowances());
        String next = null;

        if (last != null && serviceResponse.size() == limit) {
            long accountId = owner ? last.getSpender() : last.getOwner();
            var included = Map.of(ACCOUNT_ID, true, TOKEN_ID, false);
            var lastValues = Map.of(ACCOUNT_ID, accountId, TOKEN_ID, last.getTokenId());
            next = Utils.getPaginationLink(false, lastValues, included, order)
        }
```